src/Nethermind/Nethermind.Consensus/Processing/BlockchainProcessor.cs

```diff
@@ -255,54 +239,37 @@ void DecrementQueue(Hash256 blockHash, ProcessingResult processingResult, Except
         }
     }
 
-    private Task RunProcessing()
+    private async Task RunProcessing()
     {
-        TaskCompletionSource tcs = new();
+        _isMainProcessingThread.Value = IsMainProcessor;
```

Comment:
Why is this needed?

Reply:
Is used to determine if tries should be recovered from network and whether the tx pool should be notified of invalidated accounts in its cache (both should only happen from the main processor)